susemanager-utils/susemanager-sls/salt/util/mg_start_event_grains.sls

```diff
@@ -0,0 +1,6 @@
+mgr_start_event_grains:
+  file.append:
+    - name: /etc/salt/minion.d/susemanager.conf
+    - text: |
+        start_event_grains:
+          - machine_id
```

Comment:
How does this behave if grain is added to this list, but is not set for that minion? Is the grain simply ignored?

I'm asking because in the registration routine we also always check for `saltboot_initrd` grain and activation key grain as well (see https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java#L362 ).

Reply:
Yes, it's simply ignored. 

A lot of refactoring is actually possible with the registration process and I kept it limited to machine_id grain only to handle the rest of the cases separately. Now that I am thinking again, it is probably not a bad idea to refactor it further to take care of these other grains.